src/containers/StructurePage/resourceComponents/AddResourceModal.tsx

```diff
@@ -40,32 +43,73 @@ const StyledContent = styled.div`
     background-color: white;
   }
 `;
-class AddResourceModal extends Component {
-  constructor() {
-    super();
+
+interface Props {
+  onClose: () => void;
+  type: string;
+  allowPaste: boolean;
+  topicId: string;
+  topicFilters: Filter[];
+  refreshResources: () => void;
+}
+
+interface ContentType {
+  id: string;
+  metaDescription?: string;
+  title?: string;
+  imageUrl?: string;
+}
+
+interface SelectedType {
+  id: string;
+  paths?: string[];
+  title?: string;
+  url?: string;
+  metaUrl?: string;
+  description?: string;
+  coverPhotoUrl?: string;
+}
+
+interface State {
+  selected: SelectedType | null;
+  content: ContentType | null;
+  pastedUrl: string;
+  error: string | undefined | null;
+  loading: boolean;
+}
+
+type SummaryTypes = LearningPathSearchSummary | GroupSearchSummary;
+
+class AddResourceModal extends Component<Props & tType, State> {
+  constructor(props: Props & tType) {
+    super(props);
+
     this.state = {
-      selected: {},
-      article: {},
+      selected: null,
+      content: null,
       pastedUrl: '',
+      error: undefined,
+      loading: false,
     };
   }
 
-  onSelect = selected => {
+  onSelect = (selected: SelectedType) => {
     if (selected) {
-      if (selected.url && !selected.url.includes('learningpaths')) {
-        const articleId = selected.url.split('/').pop();
+      if (selected?.url && !selected?.url?.includes('learningpaths')) {
+        const articleId = Number(selected?.url?.split('/')?.pop());
```

Comment:
Ser at noen ble irritert på typescript og bare slengte på med masse `?.` for at den skulle slutte å klage :laughing: 
Tenker at siden vi sjekker på om `selected` eksisterer så kan vi fjerne optional sjekken på `selected?.url`. Samtidig kan vi fjerne `selected?.url &&` siden det er det optional chaining gjør for oss implisitt. Når vi kommer oss forbi den andre if'en så vet vi at både selected og url eksisterer så trenger ikke `?.` på de her `const articleId = Number(selected?.url?.split('/')?.pop());` den siste `?.` kan stå der siden det kan hende at strengen ikke inneholder `/`.  Men da får vi articleId = `NaN` siden `Number(undefined) = NaN` så burde kanskje ha noe fallback her. Du kan også fjerne ytterste if-sjekken siden den har vi implisitt også inn i andre if'en men da må du beholde `selected?.url` sjekken.

Reply:
Kan du utdype litt her? (Hadde satt pris på noen avsnitt neste gang :smile: )

---

> Samtidig kan vi fjerne selected?.url && siden det er det optional chaining gjør for oss implisitt.

Det vil ikke gi samme oppførsel? Siden `selected.url` er optional så vil jo da en manglende url returnere `true`.

---
> Når vi kommer oss forbi den andre if'en så vet vi at både selected og url eksisterer så trenger ikke ?. på de her const articleId = Number(selected?.url?.split('/')?.pop()); den siste ?. kan stå der siden det kan hende at strengen ikke inneholder /.

Visste ikke at kompilatoren var så smart at den ser at vi har sjekket feltene tidligere, kult! :+1: 

---

> Men da får vi articleId = NaN siden Number(undefined) = NaN så burde kanskje ha noe fallback her.

Noen forslag? La inn en isNaN check, men syns ikke det var så clean.

---

> Du kan også fjerne ytterste if-sjekken siden den har vi implisitt også inn i andre if'en men da må du beholde selected?.url sjekken.

Det vil vel også gi oss en annen oppførsel?

---